crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,305 @@
+                    $(".csh_student_answer", element).each(function(){
+                        if ($(this).find('.csh_answer_text').attr('answer') == student_answer){
+                            var html = "";
+                            var template = $('#show_no_hints').html();
+                            var data = {};
+                            html = Mustache.render(template, data);
+                            $(this).append(html);
+                            var html = "";
+                            var template = $('#add_hint_creation').html();
+                            var data = {};
+                            html = Mustache.render(template, data);
+                            $(this).append(html);
+                        }
+                    });
+                }
+                //reported hints have their corresponding answer set to "Reported"
+                else {
+                    showHintFeedback(hint, student_answer);
+                }
+              }
+            });
+            isShowingHintFeedback = true;
+        }
+    }
+
+    /**
+     * Create a text input area for the student to create a new hint. This function
+     * is triggered by clicking the "contribute a new hint" button.
+     * @param clicked is the "contribute a new hint" button that was clicked
+     */
+    function create_text_input(){ return function(clicked){
+        $('.csh_student_hint_creation', element).each(function(){
+            $(clicked.currentTarget).show();
+        });
+        $('.csh_student_text_input', element).remove();
+        $('.csh_submit_new', element).remove();
+        $(clicked.currentTarget).hide();
+        student_answer = $(clicked.currentTarget).parent().parent().find('.csh_answer_text').attr('answer');
+        $(".csh_student_answer", element).each(function(){
+            if ($('.csh_answer_text', element).attr('answer') == student_answer){
+                var html = "";
+                $(function(){
+                    var template = $('#student_hint_creation').html();
+                    var data = {student_answer: student_answer};
+                    html = Mustache.render(template, data); 
+                });
+                $(this).append(html);
+            }
+        });
+    }}
+    $(element).on('click', '.csh_student_hint_creation', create_text_input($(this)));
+
+    /**
+     * Submit a new hint created by the student to the hint pool. Hint text is in
+     * the text input area created by create_text_input. Contributed hints are specific to 
+     * incorrect answers. Triggered by clicking the "submit hint" button.
+     * @param clicked is the "submit hint" button clicked
+     */
+    function submit_new_hint(){ return function(clicked){
+    //add the newly created hint to the hinter's pool of hints
+        if($('.csh_student_text_input', element).val().length > 0){
+            var answerdata = unescape(clicked.currentTarget.attributes['answer'].value);
```

Comment:
Ignore my use of case in the code review and follow: 

     https://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml